src/api/attachments.tsx

```diff
@@ -19,8 +26,52 @@ export const usePostAttachmentMetadata = (): UseMutationResult<
   AxiosError,
   AttachmentPostMetadata
 > => {
+  const queryClient = useQueryClient();
   return useMutation({
     mutationFn: (attachmentMetadata: AttachmentPostMetadata) =>
       postAttachmentMetadata(attachmentMetadata),
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: ['Attachments'],
+      });
+    },
+  });
+};
+
+export const getAttachment = async (id: string): Promise<AttachmentPostMetadataResponse> => {
+  return storageApi.get(`/attachments/${id}`).then((response) => {
+    return response.data;
+  });
+};
+
+export const useGetAttachment = (
+  id: string
+): UseQueryResult<AttachmentPostMetadataResponse, AxiosError> => {
+  return useQuery({
+    queryKey: ['Attachment', id],
+    queryFn: () => getAttachment(id),
+  });
+};
+
+const getAttachments = async (
+  entityId: string
+): Promise<AttachmentPostMetadataResponse[]> => {
+  const queryParams = new URLSearchParams();
+  queryParams.append('entity_id', entityId);
+
+  return storageApi
+    .get(`/attachments`, {
+      params: queryParams,
+    })
+    .then((response) => response.data);
+};
+
+export const useGetAttachments = (
+  entityId?: string
+): UseQueryResult<AttachmentPostMetadataResponse[], AxiosError> => {
+  return useQuery({
+    queryKey: ['Attachments', entityId],
+    queryFn: () => getAttachments(entityId ?? ''),
+    enabled: !!entityId,
   });
```

Comment:
```suggestion
export const useGetAttachments = (
  entityId: string
): UseQueryResult<AttachmentPostMetadataResponse[], AxiosError> => {
  return useQuery({
    queryKey: ['Attachments', entityId],
    queryFn: () => getAttachments(entityId),
  });
```

If there is not a valid entityId the page would not load so it doesn't need to be optional